cocotb/share/lib/simulator/simulatormodule.c

```diff
@@ -798,6 +798,72 @@ static PyObject *get_const(PyObject *self, PyObject *args)
     return pyresult;
 }
 
+static PyObject *get_is_port(PyObject *self, PyObject *args)
+{
+    int result;
+    gpi_sim_hdl hdl;
+    PyObject *pyresult;
+
+    PyGILState_STATE gstate;
+    gstate = TAKE_GIL();
```

Comment:
There's no need to take the gil here, these functions are only callable from within python, not from the gpi code. See `get_const` for an example.

Reply:
Thanks, removed now.